### PR TITLE
Switch to modern service definition for elasticsearch-full

### PR DESCRIPTION
**Pull request by PikachuEXE**
Summary: Homebrew has officially deprecated the `plist_options` command
Example warning message: `Warning: Calling plist_options is deprecated! Use service.require_root instead.`

See https://github.com/mongodb/homebrew-brew/pull/174 for most details

### CODE REVIEW

**Comment by PikachuEXE** [comment]
I have signed the CLA, how to confirm here ._.

**Comment by PikachuEXE** [comment]
Still no one...?

**Comment by flanger001** [comment]
Bump @mgreau

**Comment by theomelo** [comment]
Bump @marclop @jasontedor

**Comment by mischabraam** [comment]
This change is needed more and more...... bump bump

**Comment by dskecse** [comment]
also closes https://github.com/elastic/homebrew-tap/issues/146

@jasontedor @mgreau is there anything left to approve and merge this PR? could you please have a look?

**Comment by drnic** [comment]
Fascinating that a company would stop work on their own homebrew tap and leave it distributing an aging version of their product (v7 vs v8). Is there some history of what happened late 2022 that led @elastic to stop working on this?

**Comment by euxx** [comment]
Before the PR be merged, we could apply this PR on locally:
```
$ cd /usr/local/Homebrew/Library/Taps/elastic/homebrew-tap
$ git fetch origin pull/144/head:patch-1
$ git checkout patch-1
```

**Review by euxx — APPROVED** [review]

**Review by ur001 — APPROVED** [review]

**Review by REFUZIION — APPROVED** [review]
Just tested these changes, and they function as expected.

**Comment by wsmi1289** [comment]
Works for me

**Comment by mischabraam** [comment]
Ok, it seems to be working.... what is the wait for?

**Comment by drnic** [comment]
They’ve abandoned this project

It’s not mentioned in the downloads docs https://www.elastic.co/downloads/elasticsearch

I guess it’s due to v8 and their move away from their OSS roots?

**Review by sdouma — APPROVED** [review]
looks good

**Comment by sdouma** [comment]
> looks good

I know this does nothing but it makes me laugh

**Comment by od-c0d3r** [comment]
LGTM, ship it

**Comment by flanger001** [comment]
They're never going to merge this. We may as well give up now.

**Comment by jaredbeck** [comment]
> It’s not mentioned in the downloads docs https://www.elastic.co/downloads/elasticsearch

Yeah, the old page (https://www.elastic.co/guide/en/elasticsearch/reference/current/brew.html) is a 404 now.

**Comment by dskecse** [comment]
Now that it's obvious this project has been abandoned and due to [this message](https://github.com/elastic/homebrew-tap/pull/126#issuecomment-1059264078) from its maintainer, I'd like to suggest anyone interested in using the fixed and updated version of ElasticSearch 7.17.14 (as opposed to [7.17.4](https://github.com/elastic/homebrew-tap/blob/main/Formula/elasticsearch-full.rb#L4) from this tap) to check out [my Homebrew tap](https://github.com/dskecse/homebrew-tap). The [`elasticsearch-full@7.17` formula](https://github.com/dskecse/homebrew-tap/blob/main/Formula/elasticsearch-full%407.17.rb) there is mostly a copy of the [`elasticsearch-full` formula](https://github.com/elastic/homebrew-tap/blob/main/Formula/elasticsearch-full.rb) from this repo except for the fixes and the version update. To install:

```
brew tap dskecse/tap
brew install elasticsearch-full@7.17
```

**Comment by mgreau** [comment]
Sorry for the radio silence here, merging.

**Review by mgreau — APPROVED** [review]
LGTM

**Comment by drnic** [comment]
Thanks @mgreau !

**Pull request merged**
